notes/reduction-passes.md

```diff
@@ -0,0 +1,225 @@
+# Reduction Passes Guide
+
+This document catalogs the reduction passes and their strategies.
```

Comment:
I think it is not helpful to have pass-by-pass information here, and those are better kept to the docstrings of the passes. Instead, this can usefully have a high level (one or two sentence) description of each module, with more detailed descriptions in the module docstrings.